### PR TITLE
feat(client): Break the monolith - preview user flow

### PR DESCRIPTION
**Pull request by abrl91**
Close: https://github.com/amplication/private-issues/issues/130

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Resource/break-the-monolith/BreakTheMonolithOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                <FlexItem direction={EnumFlexDirection.Column}>
+                  <FlexItem>
+                    <Text textStyle={EnumTextStyle.H3}>
+                      {option.displayName}
+                    </Text>
+                    <Button
+                      onClick={() => handleBreakClicked(option)}
+                      className={`${CLASS_NAME}__action-button`}
+                    >
+                      Break
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Resource/break-the-monolith/BreakTheMonolithOptions.tsx:124]: yes

File: packages/amplication-client/src/Resource/break-the-monolith/BreakTheMonolithOverviewPage.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            </Text>
+          </div>
+          <BtmButton
+            location={EnumButtonLocation.PreviewBtm}
+            openInFullScreen={false}
+            autoRedirectAfterCompletion
+            ButtonStyle={EnumButtonStyle.GradientFull}
+          />
+        </div>
+        <ModelOrganizer />
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Resource/break-the-monolith/BreakTheMonolithOverviewPage.tsx:69]: ModelOrganizer doesn't have props

File: packages/amplication-client/src/Resource/break-the-monolith/BtmButton.tsx [hunk 3]
```diff
@@ -58,33 +54,31 @@ export const BtmButton: React.FC<Props> = ({
   const onButtonSelectResource = useCallback(() => {
     setSelectedResource(currentResource);
     toggleIsOpen();
-  }, [currentResource, toggleIsOpen]);
+
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Resource/break-the-monolith/BtmButton.tsx:57]: I didn't understand

File: packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizerToolbar.tsx [hunk 4]
```diff
@@ -60,6 +62,9 @@ export default function ModelOrganizerToolbar({
   mergeNewResourcesChanges,
   resetUserAction,
 }: Props) {
+  const { currentWorkspace } = useAppContext();
+  const workspacePlan = currentWorkspace?.subscription?.subscriptionPlan;
+  const isPreviewSubscriptionPlan = isPreviewPlan(workspacePlan);
```
[the remark this answers is not in the document]
Comment (abrl91) [reply packages/amplication-client/src/Project/ArchitectureConsole/ModelOrganizerToolbar.tsx]: But I think that after the change I have made based on your next comment - the "restrictedMode" prop, we don't really need to calculate it, just send "true" or "false" (which is the default value)

**Review by yuval-hazaz — APPROVED** [review]

**Pull request merged**